scripts/skills/perks/perk_rf_kingfisher.nut

```diff
@@ -35,82 +36,92 @@ this.perk_rf_kingfisher <- ::inherit("scripts/skills/skill", {
 			return;
 
-		if (::Math.rand(1, 100) <= this.m.Chance)
+		local throwNetSkill = this.getContainer().getSkillByID("actives.throw_net");
+		if (!throwNetSkill.onVerifyTarget(this.getContainer().getActor().getTile(), _targetEntity.getTile()))
+			return;
+
+		local netItemScript = ::IO.scriptFilenameByHash(this.getContainer().getActor().getOffhandItem().ClassNameHash);
+
+		throwNetSkill.useForFree(_targetEntity.getTile());
+
+		local replacementNet = ::new(netItemScript)
+
+		this.getContainer().getActor().getItems().equip(replacementNet); // the original net is unequipped during use of throw net skill, so we equip a "new" net of the same type
+		foreach (skill in replacementNet.m.SkillPtrs)
+		{
+			// the skill isn't added to the skill_container yet, because we are within the onTargetHit function, so it goes to SkillsToAdd
+			// so we can't get it via container.getSkillByID, instead we find it by iterating over the skills of the item
+			if (skill.getID() == "actives.throw_net")
+			{
+				skill.m.IsHidden = true;
+				break;
+			}
+		}
+
+		local netEffect = _targetEntity.getSkills().getSkillByID("effects.net");
+		netEffect.m.KingfisherPerk <- ::MSU.asWeakTableRef(this);
+		local onRemoved = netEffect.onRemoved;
+		netEffect.onRemoved = function()
+		{
+			onRemoved();
+			if (!::MSU.isNull(this.m.KingfisherPerk))
+			{
+				this.m.KingfisherPerk.setSpent(false);
+			}
+		}
+		local onDeath = netEffect.onDeath;
+		netEffect.onDeath = function( _fatalityType )
 		{
-			this.setSpent(true);
-
-			local self = this;
-			_targetEntity.getSkills().add(::MSU.new("scripts/skills/effects/net_effect", function(o) {
-				o.m.KingfisherPerk <- ::MSU.asWeakTableRef(self);
-				o.onRemoved <- function()
-				{
-					if (!::MSU.isNull(this.m.KingfisherPerk))
-					{
-						this.m.KingfisherPerk.setSpent(false);
-					}
-				}
-				o.onDeath <- function( _fatalityType )
-				{
-					this.onRemoved();
-				}
-			}));
-
-			_targetEntity.getSkills().add(::MSU.new("scripts/skills/actives/break_free_skill", function(o) {
-				o.m.Icon = "skills/active_74.png";
-				o.m.IconDisabled = "skills/active_74_sw.png";
-				o.m.Overlay = "active_74";
-				o.setChanceBonus(999);
-				o.m.SoundOnUse = [
-					"sounds/combat/break_free_net_01.wav",
-					"sounds/combat/break_free_net_02.wav",
-					"sounds/combat/break_free_net_03.wav"
-				];
-			}));
-
-			local isReinforced = this.getContainer().getSkillByID("actives.throw_net").m.IsReinforced;
-			local effect = ::Tactical.spawnSpriteEffect(isReinforced ? "bust_net_02" : "bust_net", this.createColor("#ffffff"), _targetEntity.getTile(), 0, 10, 1.0, _targetEntity.getSprite("status_rooted").Scale, 100, 100, 0);
-			local flip = !_targetEntity.isAlliedWithPlayer();
-			effect.setHorizontalFlipping(flip);
-			::Time.scheduleEvent(::TimeUnit.Real, 200, this.onNetSpawn.bindenv(this), {
-				TargetEntity = _targetEntity,
-				IsReinforced = isReinforced
-			});
+			onDeath();
+			this.onRemoved();
```

Comment:
This line does not seem correct. So when the target dies then our kingfisher perk is removed?

You probably want
```
if (!::MSU.isNull(this.m.KingfisherPerk))
{
	this.m.KingfisherPerk.setSpent(false);
}
```
here

Reply:
`this` here is the `netEffect`. On death of the netted character it calls the `this.onRemoved()` function inside the netEffect which is hooked just a few lines above. That then calls the `setSpent(false)` for the kingfisrher perk.

Reply:
I wonder now if this causes potential future issues. `onRemoved` is not normally called during the `onDeath` of a skill.
What if a submod adds an additional effect to when a net_effect is removed which expects the character to still be alive

Reply:
I've edited the code to add a separate function to the skill which is now called from both `onDeath` and `onRemoved`.